### PR TITLE
Support you need during case flow corrections

### PR DESCRIPTION
**Pull request by hcarslaw**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the keep-helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.test.ts [hunk 1]
```diff
@@ -457,8 +457,8 @@ describe('ProceedingDetailsMapper', () => {
   });
 
   test('Should return value for providing a date for getDisplayDate', () => {
-    const retunrn_date = { day: undefined, month: undefined, year: 'May' };
-    expect(getDisplayDate(new Date())).toEqual(retunrn_date);
+    const retunrn_date = { day: '01', month: '01', year: '2023' };
+    expect(retunrn_date).toEqual(getDisplayDate('2023-01-01' as unknown as Date));
```
Comment (vivek-sekhar) [inline src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.test.ts:461]: Is this part of ur fix?

File: src/main/steps/common/support-you-need-during-case/summary/handler.ts [hunk 2]
```diff
@@ -107,7 +113,16 @@ function processDocSupportData(userCase: Partial<CaseWithId>) {
...
+  const partyType = getCasePartyType(userCase, userId);
+  if (partyType === PartyType.RESPONDENT) {
+    respondentUrls(urls, userCase);
+  } else if (partyType === PartyType.APPLICANT) {
+    applicantUrls(urls, userCase);
+  }
+}
+
+function respondentUrls(urls: object, userCase: CaseWithId) {
   if (userCase.reasonableAdjustments?.includes(ReasonableAdjustments.DOCUMENTS_SUPPORT)) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/support-you-need-during-case/summary/handler.ts:126]: Can u pls get rid of respondentUrls & applicantUrls methods and split each if condition as a es6 methods and then call it inside reasonableAdjustment function. Have the if conditions inside reasonableAdjustment method. So each individual method will hold a ternary operator for identifying the partyType.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.test.ts (hunk 1 above)
Comment (hcarslaw) [reply src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.test.ts]: Made change as test was failing, have removed completely now instead

File: src/main/steps/common/support-you-need-during-case/summary/handler.ts (hunk 2 above)
Comment (hcarslaw) [reply src/main/steps/common/support-you-need-during-case/summary/handler.ts:123]: Changed if conditions to arrow methods as discussed

File: src/main/steps/common/support-you-need-during-case/summary/handler.ts [hunk 3]
```diff
@@ -105,40 +115,112 @@ function processDocSupportData(userCase: Partial<CaseWithId>) {
...
-    Object.assign(urls, { parkingDetails: CA_DA_TRAVELLING_TO_COURT });
-    Object.assign(urls, { differentChairDetails: CA_DA_TRAVELLING_TO_COURT });
-    Object.assign(urls, { travellingOtherDetails: CA_DA_TRAVELLING_TO_COURT });
+    Object.assign(urls, getTravellingToCourtUrls(partyType));
   }
-}
+};
+
+const getDocumentSupportUrls = (partyType: PartyType): object => {
+  const urlObject =
```
Comment (vivek-sekhar) [inline src/main/steps/common/support-you-need-during-case/summary/handler.ts:141]: you don't require a variable, simply return directly.

File: src/main/steps/common/support-you-need-during-case/summary/handler.ts [hunk 4]
```diff
@@ -105,40 +115,112 @@ function processDocSupportData(userCase: Partial<CaseWithId>) {
...
+          courtComfort: COURT_HEARING_COMFORT,
+          lightingProvideDetails: COURT_HEARING_COMFORT,
+          otherProvideDetails: COURT_HEARING_COMFORT,
+        };
+  return urlObject;
+};
+
+const getTravellingToCourtUrls = (partyType: PartyType): object => {
+  const urlObject =
+    partyType === PartyType.RESPONDENT
```
Comment (vivek-sekhar) [inline src/main/steps/common/support-you-need-during-case/summary/handler.ts:212]: you don't require a variable, simply return directly.

File: src/main/steps/common/support-you-need-during-case/summary/handler.ts [hunk 5]
```diff
@@ -105,40 +115,112 @@ function processDocSupportData(userCase: Partial<CaseWithId>) {
...
+          familyProviderDetails: COURT_HEARING_SUPPORT,
+          therapyDetails: COURT_HEARING_SUPPORT,
+          communicationSupportOther: COURT_HEARING_SUPPORT,
+        };
+  return urlObject;
+};
+
+const getCourtHearingComfortUrls = (partyType: PartyType): object => {
+  const urlObject =
+    partyType === PartyType.RESPONDENT
```
Comment (vivek-sekhar) [inline src/main/steps/common/support-you-need-during-case/summary/handler.ts:196]: you don't require a variable, simply return directly.

File: src/main/steps/common/support-you-need-during-case/summary/handler.ts [hunk 6]
```diff
@@ -105,40 +115,112 @@ function processDocSupportData(userCase: Partial<CaseWithId>) {
...
+      : {
+          helpCommunication: COMMUNICATION_HELP,
+          signLanguageDetails: COMMUNICATION_HELP,
+          describeOtherNeed: COMMUNICATION_HELP,
+        };
+  return urlObject;
+};
+
+const getCourtHearingSupportUrls = (partyType: PartyType): object => {
+  const urlObject =
```
Comment (vivek-sekhar) [inline src/main/steps/common/support-you-need-during-case/summary/handler.ts:175]: you don't require a variable, simply return directly.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

**Pull request merged**